### PR TITLE
fix: fix typescript errors in the app

### PR DESCRIPTION
**Pull request by NabbeunNabi**
## 🔧 What changed

- Adding support for nuxt variables and plugins into typescript compiler.
- We should not have any more $t() errors!
- import from '#app' should work now. We can get vueApp variable

## 🧪 Testing instructions

- app should still work!
- you shouldn't see any more $t() errors

### CODE REVIEW

File: app.vue [hunk 1]
```diff
@@ -36,7 +36,7 @@ store.setEnableModerationPanelToTrue(currentPath)
 
 const useLocalAPI = useRuntimeConfig().public.NUXT_USE_LOCAL_API
 
-initializeGqlClient(useLocalAPI)
+initializeGqlClient(useLocalAPI as string | undefined)
```
Comment (ermish) [inline app.vue:39]: move this up to when useLocalAPI is instantiated

Comment (ermish) [inline app.vue:39]: move this up to when useLocalAPI is instantiated

File: components/ModDashboardLeftNavbar.vue [hunk 2]
```diff
@@ -36,7 +36,8 @@
 </template>
 
 <script setup lang="ts">
-import { ref, Ref } from 'vue'
+import { ref} from 'vue'
+import type { Ref } from 'vue'
```
Comment (ermish) [inline components/ModDashboardLeftNavbar.vue:40]: you can keep these in the same line, just put `type` in front of `Ref`  within the curlies

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
         "prod:start": "nuxi start",
         "prepare:ci": "nuxi prepare",
         "test": "yarn test:pinia --run && yarn test:e2e",
-        "test:pinia": "vitest",
+        "test:pinia": "nuxi prepare && nuxi generate && vitest",
```
Comment (ermish) [inline package.json:17]: I don't think we need this

File: tsconfig.json [hunk 4]
```diff
@@ -30,5 +26,7 @@
     ],
     "exclude": [
         "node_modules"
-    ]
+    ],
+    
+    "extends": "./.nuxt/tsconfig.json"
```
Comment (ermish) [inline tsconfig.json:31]: let's move this to the top of the file. It makes it easier to the next reader to know we're building on rules from a base config.

File: components/ModEditSubmissionTopbar.vue [hunk 5]
```diff
@@ -26,7 +26,8 @@
 <script setup lang="ts">
 import SVGCopyContent from '~/assets/icons/content-copy.svg'
 import SVGSuccessCheckMark from '~/assets/icons/checkmark-square.svg'
-import { Ref, ref } from 'vue';
+import {  ref } from 'vue';
```
Comment (ermish) [inline components/ModEditSubmissionTopbar.vue:29]: ♻️

File: components/ModInputField.vue [hunk 6]
```diff
@@ -41,4 +39,10 @@ const checkValidationOfInputField =(inputValue:string) => {
     isTheInputValueValid.value = props.inputValidationCheck(inputValue)
 }
 
+const handleInput = (event: Event) => {
```
Comment (ermish) [inline components/ModInputField.vue:42]: not really related to this PR context, but I'm fine with improvements 😉

Comment (ermish) [reply components/ModInputField.vue:42]: I guess unless the compiler is complaining

File: components/SearchBar.vue [hunk 7]
```diff
@@ -51,12 +51,15 @@
...
-import { ref, Ref, watchEffect } from 'vue'
+import { ref,  watchEffect } from 'vue'
 import { useSearchResultsStore } from '~/stores/searchResultsStore.js'
 import { useLocationsStore } from '~/stores/locationsStore.js'
-import { useSpecialtiesStore, SpecialtyDisplayOption } from '~/stores/specialtiesStore.js'
+import { useSpecialtiesStore } from '~/stores/specialtiesStore.js'
 import { Locale, Specialty } from "~/typedefs/gqlTypes.js"
-import { useLocaleStore, LocaleDisplay } from '~/stores/localeStore.js'
+import { useLocaleStore } from '~/stores/localeStore.js'
+import type { LocaleDisplay } from '~/stores/localeStore.js'
```
Comment (ermish) [inline components/SearchBar.vue:60]: ♻️

File: components/SubmissionForm.vue [hunk 8]
```diff
@@ -93,7 +93,8 @@
 import { useSubmissionStore } from "~/stores/submissionStore"
 import { useLocaleStore } from "~/stores/localeStore"
 import * as validations from '../utils/formValidations'
-import { computed, ComputedRef } from 'vue'
+import { computed } from 'vue'
```
Comment (ermish) [inline components/SubmissionForm.vue:96]: ♻️

File: package.json (hunk 3 above)
Comment (ermish) [reply package.json:17]: oh n/m I guess we do.

**Review by ermish — CHANGES_REQUESTED** [review]
it works! a few cleanups
this duplicates #560 so I'll close that one in favor of this

File: app.vue (hunk 1 above)
Comment (ermish) [reply app.vue:39]: best practice for types: 
Add `as` casts when you want to change the type to something different or declare a new variable. If the variable is supposed to be of that type, put it in that line instead so it's more visible that it was expected to be of that type.

File: components/ModInputField.vue (hunk 6 above)
Comment (NabbeunNabi) [reply components/ModInputField.vue:42]: The typescript compiler was complaining about the type so this was the best practice I could find to fix it

File: app.vue [hunk 9]
```diff
@@ -24,18 +24,17 @@
 import { initializeGqlClient } from './utils/graphql.js'
 import { useNuxtApp } from "#app";
 import { useRuntimeConfig } from '#imports'
-const { $viewport } = useNuxtApp();
 import { useModerationScreenStore } from "~/stores/moderationScreenStore"
 import { useRoute } from 'vue-router'
+const { $viewport } = useNuxtApp();
```
Comment (ermish) [inline app.vue:29]: there's a semicolon added

File: tsconfig.json [hunk 10]
```diff
@@ -30,5 +27,5 @@
     ],
     "exclude": [
         "node_modules"
-    ]
+    ],
```
Comment (ermish) [inline tsconfig.json:30]: added unnecessary comma

**Review by ermish — APPROVED** [review]
looks good! :shipit:

File: components/ModInputField.vue [hunk 11]
```diff
@@ -10,18 +10,15 @@
                 :placeholder='placeholder'
                 :required='required'
                 :value="inputValue"
-                @input="event => {
-                    inputValue = event.target.value
-                    store.setInputField(label, inputValue)
-                }"
+                @input="handleInput"
```
Comment (theyokohamalife) [inline components/ModInputField.vue:13]: Nice! I was mentioning to @Anissa3005 that this should be moved to a function in the script.

**Pull request merged**